### PR TITLE
SHAPRFECV speedup for bigger use-cases and some simple refactoring.

### PR DESCRIPTION
**Pull request by ReinierKoops**
This PR depends on the PR to be accepted: https://github.com/ing-bank/probatus/pull/248

______

This cleanup removes some more unused code and simplifies parts of our implementations. It should allow for a boost in performance for bigger use-cases, although minimal.

Also fix:

- [x] #242 comments 
- [x] #255
- [x] #245

### CODE REVIEW

File: probatus/interpret/model_interpret.py [hunk 1]
```diff
@@ -86,7 +86,8 @@ def __init__(self, model, scoring="roc_auc", verbose=0, random_state=None):
                 reproducible results set it to an integer.
         """
         self.model = model
-        self.scorer = get_single_scorer(scoring)
+        self.scoring = scoring  # (str) name of the metric
```
Comment (ReinierKoops) [inline probatus/interpret/model_interpret.py:89]: Add check if not a str

**Comment by ReinierKoops** [comment]
Depends on https://github.com/ing-bank/probatus/pull/248 before it can be merged.

**Comment by ReinierKoops** [comment]
Please fix #242 or comments and #255 in this pr

**Comment by ReinierKoops** [comment]
Also replace scorer with sklearn implementation #245

**Review by adri0 — APPROVED** [review]
LGTM

File: tests/feature_elimination/test_feature_elimination.py [hunk 2]
```diff
@@ -628,6 +551,7 @@ def test_LightGBM_stratified_kfold(random_state):
     X[0] = X[0].astype("float")
     y = [0] * 5 + [1] * 5
 
+    model = LGBMClassifier(random_state=random_state)
     model = LGBMClassifier(random_state=random_state)
```
[the remark this answers is not in the document]
Comment (ReinierKoops) [reply tests/feature_elimination/test_feature_elimination.py]: solved

**Pull request merged**